src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -36,7 +38,8 @@
     private final double transmissionThresholdRadius;
 
     private final HashMap<String, Double> distancesStats = new HashMap<>();
-    private final HashMap<String, Double> beaconsObservedSum = new HashMap<>();
+    private final HashMap<Beacon, observedInterval.Builder> beaconsObservedOpenInterval = new HashMap<>();
+    private final HashMap<Beacon, List<observedInterval>> beaconsObservedIntervals = new HashMap<>();
```

Comment:
Consider rename it to `beaconsObservedCurrentInterval` or add a comment instead.